test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/FrameworkAssemblyResolverTests.cs

```diff
@@ -0,0 +1,322 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Versioning;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.PackageManagement.VisualStudio.Test
+{
+    public class FrameworkAssemblyResolverTests : IClassFixture<TestDirectoryFixture>
+    {
+        private readonly ConcurrentDictionary<string, List<FrameworkAssembly>> _dictionary;
+        private readonly TestDirectoryFixture _fixture;
+
+        private static readonly FrameworkName _frameworkName = new FrameworkName(".NETFramework,Version=v4.5.2");
```

Comment:
I do not want unit tests depending on what .NET framework version is installed.  These tests mock out a partial .NET Framework reference assemblies installation, enough to validate basic scenarios.  The E2E test does verify the whole scenario.  If you still feel that this needs more E2E coverage, let me know.